### PR TITLE
Feat/no permission ineligible reason

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

The new ineligibility reason "no-permission" was introduced. Because of the sns neuron can be ineligible because of permission absence.

# Changes

- new type `NeuronIneligibilityReason`
- util function `snsNeuronsIneligibilityReasons`
- add new reason support to the `<IneligibleCard>`
- "dev-remove-permissions" dev feature (needs for testing)

# Tests

- update ineligibility related tests
- snsNeuronsIneligibilityReasons
- IneligibleCard supports new reason

# Screenshots

image

## Remove Permissions modal

image

image

### CODE REVIEW

File: frontend/src/lib/modals/sns/neurons/AddPermissionsModal.svelte [hunk 1]
```diff
@@ -19,14 +22,16 @@
...
   export let rootCanisterId: Principal;
   export let reloadNeuron: () => Promise<void>;
+  export let mode: "add" | "remove";
 
+  let modeLabel = mode === "add" ? "Add" : "Remove";
   const steps: WizardSteps = [
     {
-      name: "AddPrincipal",
-      title: "Add Principal",
+      name: `Principal`,
```
Comment (lmuntaner) [inline frontend/src/lib/modals/sns/neurons/AddPermissionsModal.svelte:30]: Why do you use the backtick "`" for this name?

File: frontend/src/lib/modals/sns/neurons/AddPermissionsModal.svelte [hunk 2]
```diff
@@ -19,14 +22,16 @@
...
+      title: `${modeLabel} Principal`,
     },
     {
-      name: "AddPermissions",
+      name: `Permissions`,
```
Comment (lmuntaner) [inline frontend/src/lib/modals/sns/neurons/AddPermissionsModal.svelte:34]: Ditto

**Review by lmuntaner — APPROVED** [review]
Thanks!!! Looks good to me, just a couple of questions!

File: frontend/src/lib/modals/sns/neurons/AddPermissionsModal.svelte (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/modals/sns/neurons/AddPermissionsModal.svelte]: Good catch! It was dynamic in an intermediate version. Fixed.

**Pull request merged**